### DISCUSSION TITLE
Input required

### DISCUSSION THREAD

**Discussion by StuartMcD** [solspace/craft-freeform#445] [Questions & Answers] [answered]
Building a form by hand, using a mix of hand code and {{field.renderInput()}} Where renderInput is used and the field in the CMS is marked as 'required' - it doesn't add the html `required` attribute :(

Could be another same-but-different issue with radio groups - set one of two options to 'checked' and it doesn't honor on the front end - again using the {{field.renderInput()}} tag.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.2.1
- Freeform Version: 4.0.0
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Comment by kjmartens** [comment]
Sorry for the delay @StuartMcD,

By default, Freeform will add `data-required` attribute in order to work with the built-in AJAX validation. However, if you want the `required` attribute, you need to add the `useRequiredAttribute: true` parameter, either at form level (to affect all fields) or field level:

```
{{ field.renderInput({ useRequiredAttribute: true }) }}
```

Hope this helps! 🙂

**Reply by StuartMcD** [reply]
I believe this has stopped working. I'm manual rendering a form and can use it in the form or on the renderInput 

`{{field.renderInput({useRequiredAttribute: true}) }}`

`  attributes: {
    ....
  },
  useRequiredAttribute: true,
`

Makes no difference. All fields are required in the form builder.

If I use the `form.render()` approach, required attribute is added to all fields.

Using: 
Craft 4.8.6
FF 5.1.8

Update not new install.

**Comment by kjmartens** [comment, marked as answer]
@StuartMcD In Freeform 5, this has changed. Using the new [Template Overrides](https://docs.solspace.com/craft/freeform/v5/templates/formatting/#template-overrides) feature, you can do this instead: 🙂 

``` twig
{{ form.render({
    fields: {
        ":required": {
            attributes: {
                label: { "+class": "label-required" },
                input: { "required": true },
            },
        },
    }
}) }}
```

https://docs.solspace.com/craft/freeform/v5/guides/required-attribute/
